x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -141,4 +206,176 @@ public void removeFromCache(final Predicate<CacheKey> predicate) {
         }
         cache.refresh();
     }
+
+    // used in tests
+    CacheSynchronizationTask getCacheSyncTask() {
+        return cacheSyncTask;
+    }
+
+    void setCacheSyncInterval(TimeValue interval) {
+        assert cacheSyncTask != null;
+        cacheSyncTask.setInterval(interval);
+    }
+
+    /**
+     * Reschedule the {@link CacheSynchronizationTask} if the local data node is hosting searchable snapshot shards.
+     */
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        assert cacheSyncTask != null;
+
+        if (event.routingTableChanged()) {
+            final ClusterState clusterState = event.state();
+            final DiscoveryNode localNode = clusterState.getNodes().getLocalNode();
+            assert localNode.isDataNode();
+
+            final boolean shouldSynchronize = hasSearchableSnapshotShards(clusterState, localNode.getId());
```

Comment:
I wonder if we need to couple this to cluster state updates? Instead we could either trigger on the presence of any cache files or as proposed in another comment an explicit fsync-needed registration?